src/core/dependenciesMintPolicy.js

```diff
@@ -0,0 +1,133 @@
+ * a project should be paid via CredRank and the graph-based system, rather
+ * than "out-of-band" fiat/configuration like the mint policies in this module.
+ * Thus, we're naming it the dependencyMintPolicy to communicate that
+ * intention.
+ */
+import {type NodeAddressT, NodeAddress} from "./graph";
+import {type TimestampMs} from "../util/timestamp";
+
+export type DependencyMintPolicy = {|
+  // The node address that will receieve the extra minted Cred
+  +address: NodeAddressT,
+  // Information on how the minted Cred varies in time.
+  +periods: $ReadOnlyArray<DependencyMintPeriod>,
+|};
+
+export type DependencyMintPeriod = {|
+  // What proportion of the project's raw Cred should be minted to this
+  // dependency during this period. For example, if the weight is 0.05, and the
+  // project has a pre-dependencies total Cred of 1000 in a given interval,
+  // then this dependench will receive an additional 50 Cred.
+  +weight: number,
+  // Timestamp or number because we allow -Infinity as the first "timestamp"
+  // Within any policy, the period timestamps must be in sorted order, or else
+  // an error will be thrown.
+  +startTimeMs: TimestampMs | number,
+|};
+
+/**
+ * The ProcessedMintPolicy is a DependencyMintPolicy which has
+ * been transformed so that it matches the abstractions available when
+ * we're doing raw cred computation: instead of an address, we track an index
+ * into the canonical node order, and rather than arbitrary client-provided
+ * periods, we compute the weight for each Interval.
+ */
+export type ProcessedMintPolicy = {|
+  +nodeIndex: number,
+  +intervalWeights: $ReadOnlyArray<number>,
+|};
+
+export function processMintPolicy(
+  policy: DependencyMintPolicy,
+  nodeOrder: $ReadOnlyArray<NodeAddressT>,
+  intervalStarts: $ReadOnlyArray<TimestampMs>
+): ProcessedMintPolicy {
+  const {address, periods} = policy;
+  const nodeIndex = nodeOrder.indexOf(address);
+  if (nodeIndex === -1) {
+    throw new Error(
+      `address not in nodeOrder: ${NodeAddress.toString(address)}`
+    );
+  }
+  const intervalWeights = _alignPeriodsToIntervals(periods, intervalStarts);
+  return {nodeIndex, intervalWeights};
+}
+
+export function _alignPeriodsToIntervals(
+  periods: $ReadOnlyArray<DependencyMintPeriod>,
+  intervalStarts: $ReadOnlyArray<TimestampMs>
+): $ReadOnlyArray<number> {
+  if (periods.length === 0) {
+    return intervalStarts.map(() => 0);
+  }
+  // Validate the periods to make sure they are in order and the weights
+  // are finite and non-negative
+  let currentStartTimeMs = periods[0].startTimeMs;
+  for (const {startTimeMs, weight} of periods) {
+    if (currentStartTimeMs > startTimeMs) {
+      throw new Error(
+        `mint periods out of order: ${currentStartTimeMs} > ${startTimeMs}`
+      );
+    }
+    currentStartTimeMs = startTimeMs;
+    if (weight < 0 || !Number.isFinite(weight)) {
+      throw new Error(`invalid mint weight: ${weight}`);
+    }
+  }
```

Comment:
Wondering if this is something we can make a parser for instead of manually validating like this.